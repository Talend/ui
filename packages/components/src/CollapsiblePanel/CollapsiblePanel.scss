$tc-collapsible-panel-padding-smaller: $padding-smaller !default;
$tc-collapsible-panel-padding-normal: $padding-normal !default;
$tc-collapsible-panel-padding-larger: $padding-larger !default;
$tc-collapsible-panel-btn-color: #555964;
$tc-collapsible-descriptive-panel-content-description-color: #808080 !default;
$tc-toggle-color: #555964 !default;

@mixin type-col($width: 100%) {
	flex-basis: $width;
	display: flex;
	overflow: hidden;

	> span {
		white-space: nowrap;
		overflow: hidden;
		text-overflow: ellipsis;
	}
}

@mixin col-style {
	> :global(.btn) {
		line-height: 1em;
		color: $tc-collapsible-panel-btn-color;
	}

	&.col-4 {
		@include type-col(25%);
	}

	&.col-5 {
		@include type-col(20%);
	}

	&.col-6 {
		@include type-col(calc(100% / 6));
	}

	&.col-7 {
		@include type-col(calc(100% / 7));
	}

	&.col-8 {
		@include type-col(calc(100% / 8));
	}

	&.col-9 {
		@include type-col(calc(100% / 9));
	}

	&.col-10 {
		@include type-col(calc(10%));
	}
}

// due to react-bootstrap
:global(.panel .panel) {
	margin: 0;
	border: none;
}

.tc-collapsible-panel {
	position: relative;

	&.selected {
		border-left: 4px solid $brand-primary;
		padding-left: 1px;
	}

	&.info,
	&.success,
	&.danger,
	&.muted,
	&.warning {
		padding-left: 1px;
		border-left: 4px solid;
	}

	&.info {
		border-left-color: $brand-info;
	}

	&.success {
		border-left-color: $brand-success;
	}

	&.danger {
		border-left-color: $brand-danger;
	}

	&.muted {
		border-left-color: $gray;
	}

	&.warning {
		border-left-color: $brand-warning;
	}

	.toggle svg {
		transform-origin: center;
	}

	&.open {
		.toggle svg {
			transform: rotate(180deg);
		}
	}

	.group {
		display: flex;
	}

	.panel-heading {
		display: flex;
		justify-content: space-between;
		padding: 0;

		> * {
			padding: 0;
			letter-spacing: 0;
			text-decoration: none;
		}

		.panel-title {
			display: flex;
			align-items: center;
			justify-content: space-between;
			flex-basis: 100%;
			overflow: hidden;
			text-decoration: none;
			text-transform: none;
			font-size: 1.4rem;
			padding-left: 5px;
			padding-right: 5px;

			> * {
				@include col-style;
				display: flex;
				align-items: center;
				height: 4rem;
			}
		}
	}
}

.default-panel {
	.panel-heading {
		.panel-title {
			flex-basis: calc(100% - 35px);

			:global(.label) {
				padding: 0.3em 0.6em;
			}
		}
	}

	:global(.label) {
		white-space: nowrap;
		text-overflow: ellipsis;
		overflow: hidden;
	}

	.group {
		justify-content: space-between;
		padding-right: 15px;
	}

	:global(.panel-body) {
		max-height: 50vh;
		overflow: auto;
		word-break: break-all;

		.content {
			display: flex;
			flex-wrap: wrap;
			width: 100%;

			.label {
				flex-basis: 10%;
				flex-shrink: 0;
				flex-grow: 1;
			}

			.description {
				flex-basis: 90%;
				flex-shrink: 0;
				flex-grow: 1;
				font-weight: normal;
			}
		}
	}
}

.descriptive-panel {
	border-radius: 0;
	margin-bottom: 0;
	box-shadow: none;
	border: none;
	padding-left: 5px;

	:global(.panel) {
		box-shadow: none;
	}

	:global(.toggle) {
		svg {
			fill: $tc-toggle-color;
			height: 10px;
			width: 10px;
		}
	}

	&.selected {
		.title,
		.detail {
			color: $brand-primary;
		}
	}

	:global(.panel-heading) {
		background-color: white;
		border-color: transparent;
	}

	.group {
		justify-content: flex-start;
	}

	.title {
		text-transform: none;
		font-weight: bold;
		overflow: hidden;
		text-overflow: ellipsis;
		white-space: nowrap;
	}

	.tag {
		margin-left: 2px;
		white-space: nowrap;
		font-size: 12px;
	}

	.detail {
		white-space: nowrap;
		justify-content: flex-end;
	}

	:global(.panel-body) {
		padding: 10px;
		padding-top: 0;

		.content {
			display: flex;
			flex-direction: column;

			.head {
				display: flex;
				justify-content: space-between;

				> span {
					overflow: hidden;
					text-overflow: ellipsis;
					white-space: nowrap;
				}
			}

			.content-description {
				white-space: pre-wrap;
				padding-top: 5px;
				color: $tc-collapsible-descriptive-panel-content-description-color;
			}
		}
	}
}
