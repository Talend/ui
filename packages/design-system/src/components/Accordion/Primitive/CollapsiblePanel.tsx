import { forwardRef, ReactChild, Ref, useState, useEffect, HTMLAttributes } from 'react';
import classNames from 'classnames';

import { useId } from '../../../useId';
import { DataAttributes } from '../../../types';
import { variants } from '../../Status/Primitive/StatusPrimitive';

import CollapsiblePanelHeader from './CollapsiblePanelHeader';
import { PanelHeaderAction } from './types';
import styles from './CollapsiblePanel.module.scss';

type CollapsiblePanelCommonPropsType = {
	children: ReactChild;
	managed?: boolean;
	expanded?: boolean;
	index?: number;
	action?: PanelHeaderAction;
	size?: 'S' | 'M';
	metadata?: ReactChild[];
	isFirst?: boolean;
	isLast?: boolean;
	disabled?: boolean;
	onToggleExpanded?: (index: number) => void;
} & Omit<HTMLAttributes<HTMLDivElement>, 'className' | 'style'> &
	DataAttributes;

type CollapsiblePanelWithTitlePropsType = {
	title: ReactChild;
	status?: never;
};

type CollapsiblePanelWithStatusPropsType = {
	title?: never;
	status: keyof typeof variants;
};

export type CollapsiblePanelProps = CollapsiblePanelCommonPropsType &
	(CollapsiblePanelWithTitlePropsType | CollapsiblePanelWithStatusPropsType);

export const CollapsiblePanel = forwardRef(
	(
		{
			id,
			children,
			managed,
			expanded,
			index,
			onToggleExpanded,
			title,
			status,
			action,
			size,
			metadata,
			isFirst = false,
			isLast = false,
			disabled = false,
			...rest
		}: CollapsiblePanelProps,
		ref: Ref<HTMLButtonElement>,
	) => {
		const [localExpanded, setLocalExpanded] = useState(!!expanded);

		const componentId = useId(id);
		const controlId = `CollapsiblePanel__control--${componentId}`;
		const sectionId = `CollapsiblePanel__content--${componentId}`;

		useEffect(() => {
			if (managed && expanded !== localExpanded) {
				setLocalExpanded(!!expanded);
			}
		}, [expanded, managed, localExpanded]);

		function handleToggleExpanded() {
			if (onToggleExpanded && managed && (index || index === 0)) {
				onToggleExpanded(index);
			} else {
				setLocalExpanded(!localExpanded);
			}
		}

		return (
			<div
				className={classNames(styles.panelWrapper, {
					[styles.panelWrapper__alone]: !managed,
					[styles.panelWrapper__first]: isFirst,
					[styles.panelWrapper__last]: isLast,
					[styles.panelWrapper__notLast]: managed && !isLast,
				})}
				{...rest}
			>
				<CollapsiblePanelHeader
					controlId={controlId}
					sectionId={sectionId}
					expanded={!disabled && localExpanded}
					handleClick={handleToggleExpanded}
					title={title}
					status={status}
					action={action}
					metadata={metadata}
					size={size}
					disabled={disabled}
					ref={ref}
				/>
				{localExpanded && (
					<div
						id={sectionId}
						role="region"
						aria-labelledby={controlId}
						className={styles.panelContent}
						data-test="panel.section"
						data-testid="panel.section"
					>
						{children}
					</div>
				)}
			</div>
		);
	},
);

CollapsiblePanel.displayName = 'CollapsiblePanel';
