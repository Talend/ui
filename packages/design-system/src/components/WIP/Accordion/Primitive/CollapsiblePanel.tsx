import React, { forwardRef, ReactChild, Ref, useState, useEffect, HTMLAttributes } from 'react';
import classNames from 'classnames';
import { unstable_useId as useId } from 'reakit/Id';

import { DataAttributes } from '../../../../types';

import CollapsiblePanelHeader from './CollapsiblePanelHeader';
import { PanelHeaderAction } from './types';
import styles from './CollapsiblePanel.module.scss';

export type CollapsiblePanelPropsType = {
	children: ReactChild;
	managed?: boolean;
	expanded?: boolean;
	index?: number;
	title: string;
	action?: PanelHeaderAction;
	size?: 'S' | 'M';
	metadata?: ReactChild[];
	isFirst?: boolean;
	isLast?: boolean;
	disabled?: boolean;
	onToggleExpanded?: (index: number) => void;
} & Omit<HTMLAttributes<HTMLDivElement>, 'className' | 'style'> &
	DataAttributes;

const CollapsiblePanel = forwardRef(
	(
		{
			id,
			children,
			managed,
			expanded,
			index,
			onToggleExpanded,
			title,
			action,
			size,
			metadata,
			isFirst = false,
			isLast = false,
			disabled = false,
			...rest
		}: CollapsiblePanelPropsType,
		ref: Ref<HTMLButtonElement>,
	) => {
		const [localExpanded, setLocalExpanded] = useState(!!expanded);

		const { id: reakitId } = useId();
		const componentId = id || reakitId;
		const controlId = `CollapsiblePanel__control--${componentId}`;
		const sectionId = `CollapsiblePanel__content--${componentId}`;

		useEffect(() => {
			if (managed && expanded !== localExpanded) {
				setLocalExpanded(!!expanded);
			}
		}, [expanded, managed, localExpanded]);

		function handleToggleExpanded() {
			if (onToggleExpanded && managed && (index || index === 0)) {
				onToggleExpanded(index);
			} else {
				setLocalExpanded(!localExpanded);
			}
		}

		return (
			<div
				className={classNames(styles.panelWrapper, {
					[styles.panelWrapper__alone]: !managed,
					[styles.panelWrapper__first]: isFirst,
					[styles.panelWrapper__last]: isLast,
					[styles.panelWrapper__notLast]: managed && !isLast,
				})}
				{...rest}
			>
				<CollapsiblePanelHeader
					controlId={controlId}
					sectionId={sectionId}
					expanded={!disabled && localExpanded}
					handleClick={handleToggleExpanded}
					title={title}
					action={action}
					metadata={metadata}
					size={size}
					disabled={disabled}
					ref={ref}
				/>
				{localExpanded && (
					<div
						id={sectionId}
						role="region"
						aria-labelledby={controlId}
						className={styles.panelContent}
						data-test="panel.section"
					>
						{children}
					</div>
				)}
			</div>
		);
	},
);

CollapsiblePanel.displayName = 'CollapsiblePanel';

export default CollapsiblePanel;
