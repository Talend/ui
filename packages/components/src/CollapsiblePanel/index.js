import CollapsiblePanel, { TYPE_ACTION } from './CollapsiblePanel.component';

export default CollapsiblePanel;
export { TYPE_ACTION };
