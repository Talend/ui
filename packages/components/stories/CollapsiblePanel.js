import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { checkA11y } from '@storybook/addon-a11y';
import talendIcons from '@talend/icons/dist/react';

import { CollapsiblePanel, IconsProvider } from '../src/index';

const icons = {
	'talend-cross': talendIcons['talend-cross'],
	'talend-download': talendIcons['talend-download'],
	'talend-check': talendIcons['talend-check'],
	'talend-caret-down': talendIcons['talend-caret-down'],
	'talend-warning': talendIcons['talend-warning'],
};

const content = [
	{
		label: 'Content1',
		description: 'Description1',
	},
	{
		label: 'Content2',
		description: 'Description2',
	},
];

const status = {
	displayMode: 'status',
	status: 'inProgress',
	label: 'In Progress',
	actions: [
		{
			label: 'cancel',
			icon: 'talend-cross',
			onClick: action('onCancel'),
			bsSize: 'small',
		},
	],
};

const button = {
	displayMode: 'action',
	label: 'Download',
	icon: 'talend-download',
	onClick: action('onDownload'),
	hideLabel: true,
	link: true,
};

const label1 = {
	label: 'by Charles',
	bsStyle: 'default',
	tooltipPlacement: 'top',
};

const label2 = {
	label: 'on Talend Data Preparation Server',
	bsStyle: 'default',
	tooltipPlacement: 'top',
};

const label3 = {
	displayMode: 'badge',
	label: 'XML',
	bsStyle: 'info',
	tooltipPlacement: 'top',
};

const label4 = {
	label: 'on Hadoop Cluster',
	bsStyle: 'default',
	tooltipPlacement: 'top',
};

const propsPanel = {
	header: [status, label1, label2, button, label3],
};

const propsPanelWithActions = {
	header: [
		{ ...status, actions: [], status: 'successful', label: 'Successful', icon: 'talend-check' },
		label1,
		label4,
		button,
		label3,
	],
	content,
	onToggle: action('onToggle'),
};

const propsPanelWithoutActions = {
	header: [
		{ ...status, actions: [], status: 'canceled', label: 'Canceled', icon: 'talend-cross' },
		label1,
		label2,
		{},
		label3,
	],
};

const propsCollapsiblePanel = {
	header: [
		{ ...status, status: 'failed', label: 'Failed', icon: 'talend-cross' },
		label1,
		label4,
		button,
		label3,
	],
	content,
	onToggle: action('onToggle'),
};

const propsInProgressCollapsiblePanel = {
	header: [status, label1, label4, button, label3],
	content,
	onToggle: action('onToggle'),
};

const propsFailedCollapsiblePanel = {
	header: [
		{ ...status, status: 'failed', label: 'Failed', icon: 'talend-cross' },
		label1,
		label4,
		button,
		label3,
	],
	content,
	onToggle: action('onToggle'),
};

const propsSuccessfulCollapsiblePanel = {
	header: [
		{ ...status, status: 'successful', label: 'Successful', icon: 'talend-check' },
		label1,
		label4,
		button,
		label3,
	],
	content,
	onToggle: action('onToggle'),
};
const propsWarningCollapsiblePanel = {
	header: [
		{ ...status, status: 'warning', label: 'Warning', icon: 'talend-warning' },
		label1,
		label4,
		button,
		label3,
	],
	content,
	onToggle: action('onToggle'),
};

const propsCanceledCollapsiblePanel = {
	header: [
		{ ...status, status: 'canceled', label: 'Canceled', icon: 'talend-cross' },
		label1,
		label4,
		button,
		label3,
	],
	content,
	onToggle: action('onToggle'),
};

const propsCollapsiblePanelWithHeaderGroups = {
	header: [
		{ ...status, status: 'failed', label: 'Failed', icon: 'talend-cross' },
		label1,
		label4,
		[button, label3],
	],
	content,
	onToggle: action('onToggle'),
	expanded: true,
};

const propsCollapsiblePanelWithHeaderGroupsWithProgress = {
	header: [{ ...status, progress: '70' }, label1, label4, [button, label3]],
	content,
	onToggle: action('onToggle'),
	expanded: true,
};

const version1 = {
	label: 'Version 1 Version 1 Version 1 Version 1',
	bsStyle: 'default',
	tooltipPlacement: 'top',
	className: 'title',
};
const readOnlyLabel = {
	label: '(Read Only)',
	bsStyle: 'default',
	tooltipPlacement: 'top',
	className: 'tag',
};
const timeStamp = {
	label: '05/02/2017 14:44:55',
	bsStyle: 'default',
	tooltipPlacement: 'top',
	className: 'detail',
};

const propsCollapsibleSelectablePanel = {
	header: [[version1, readOnlyLabel], timeStamp],
	content: {
		head: [
			{
				label: '21 steps',
				bsStyle: 'default',
				tooltipPlacement: 'top',
			},
			{
				label: 'by Henry-Mayeul de Benque',
				bsStyle: 'default',
				tooltipPlacement: 'top',
				className: 'text-right',
			},
		],
		description: `1 - First row, ...\n2 - Second row, ...\n3 - Third row, ...\nLast row, ...`,
	},
	onSelect: action('onselect'),
	onToggle: action('onToggle'),
	theme: 'descriptive-panel',
};

const propsCollapsibleSelectedPanel = {
	...propsCollapsibleSelectablePanel,
	expanded: true,
	status: 'selected',
	theme: 'descriptive-panel',
};

const propsSelectedPanelWithoutContent = {
	header: [[version1, readOnlyLabel], timeStamp],
	onSelect: action('onselect'),
	onToggle: action('onToggle'),
	expanded: true,
	status: 'selected',
	theme: 'descriptive-panel',
};

const propsCollapsibleSelectablePanelWithoutTag = {
	...propsSelectedPanelWithoutContent,
	header: [version1, timeStamp],
	theme: 'descriptive-panel',
};

storiesOf('CollapsiblePanel', module)
	.addDecorator(checkA11y)
	.addWithInfo('Key/Value content', () => (
		<div className="col-lg-offset-1 col-lg-10">
			<IconsProvider defaultIcons={icons} />
			<p>By default :</p>
			<div id="default">
				<CollapsiblePanel {...propsCollapsiblePanel} />
			</div>
			<p>No content :</p>
			<div id="no-content">
				<CollapsiblePanel {...propsPanel} />
			</div>
			<p>No content - Header with Actions :</p>
			<div id="with-actions">
				<CollapsiblePanel {...propsPanelWithActions} />
			</div>
			<p>No content - Header without Actions :</p>
			<div id="without-actions">
				<CollapsiblePanel {...propsPanelWithoutActions} />
			</div>
			<p>Header with groups :</p>
			<div id="with-groups">
				<CollapsiblePanel {...propsCollapsiblePanelWithHeaderGroups} />
			</div>
			<p>Header with fixed circularProgress :</p>
			<div id="with-fixed-progress">
				<CollapsiblePanel {...propsCollapsiblePanelWithHeaderGroupsWithProgress} />
			</div>
		</div>
	))
	.addWithInfo('Textual content', () => (
		<div className="col-lg-offset-1 col-lg-10">
			<IconsProvider defaultIcons={icons} />
			<p>With Content:</p>
			<div id="textual-with-content">
				<CollapsiblePanel {...propsCollapsibleSelectablePanel} />
			</div>
			<p>Opened with Content:</p>
			<div id="textual-opened-with-content">
				<CollapsiblePanel {...propsCollapsibleSelectablePanel} expanded />
			</div>
			<p>Panel Without readonly tag and without content:</p>
			<div id="textual-without-readonly">
				<CollapsiblePanel {...propsCollapsibleSelectablePanelWithoutTag} status={''} />
			</div>
		</div>
	))
	.addWithInfo('Selected Collapsible', () => (
		<div className="col-lg-offset-1 col-lg-10">
			<IconsProvider defaultIcons={icons} />
			<p>Selected Panel With Content:</p>
			<div id="selected-with-content">
				<CollapsiblePanel {...propsCollapsibleSelectedPanel} />
			</div>
			<p>Selected Panel Without readonly tag and without content:</p>
			<div id="selected-without-readonly">
				<CollapsiblePanel {...propsCollapsibleSelectablePanelWithoutTag} />
			</div>
			<p>Selected Panel Without content:</p>
			<div id="selected-without-content">
				<CollapsiblePanel {...propsSelectedPanelWithoutContent} />
			</div>
			<p>Selected key/Value CollapsiblePanel:</p>
			<div id="selected-key-value">
				<CollapsiblePanel {...propsCollapsiblePanel} status={'selected'} />
			</div>
			<p>Selected key/Value CollapsiblePanel without content:</p>
			<CollapsiblePanel {...propsPanelWithoutActions} status={'selected'} />
		</div>
	))
	.addWithInfo('Status Collapsible', () => (
		<div className="col-lg-offset-1 col-lg-10">
			<IconsProvider defaultIcons={icons} />
			<p>CollapsiblePanel with status info:</p>
			<div id="status-info">
				<CollapsiblePanel {...propsInProgressCollapsiblePanel} status={'inProgress'} />
			</div>
			<p>CollapsiblePanel with status successful:</p>
			<div id="status-success">
				<CollapsiblePanel {...propsSuccessfulCollapsiblePanel} status={'successful'} />
			</div>
			<p>CollapsiblePanel with status failed:</p>
			<div id="status-failed">
				<CollapsiblePanel {...propsFailedCollapsiblePanel} status={'failed'} />
			</div>
			<p>CollapsiblePanel with status warning:</p>
			<div id="status-warning">
				<CollapsiblePanel {...propsWarningCollapsiblePanel} status={'warning'} />
			</div>
			<p>CollapsiblePanel with status canceled:</p>
			<div id="status-canceled">
				<CollapsiblePanel {...propsCanceledCollapsiblePanel} status={'canceled'} />
			</div>
		</div>
	))
	.addWithInfo('Adjustment textual collapsibles', () => (
		<div className="col-lg-offset-1 col-lg-10">
			<IconsProvider defaultIcons={icons} />
			<p>Adjustement Story:</p>
			<div id="selectable-panel-with-content1">
				<CollapsiblePanel {...propsCollapsibleSelectedPanel} />
			</div>
			<div id="selectable-panel-with-content2">
				<CollapsiblePanel {...propsCollapsibleSelectablePanel} expanded />
			</div>
			<div id="selectable-panel-with-content3">
				<CollapsiblePanel {...propsCollapsibleSelectablePanelWithoutTag} />
			</div>
			<div id="selectable-panel-with-content4">
				<CollapsiblePanel {...propsCollapsibleSelectablePanel} />
			</div>
			<div id="selected-panel-without-content5">
				<CollapsiblePanel {...propsSelectedPanelWithoutContent} />
			</div>
		</div>
	));
