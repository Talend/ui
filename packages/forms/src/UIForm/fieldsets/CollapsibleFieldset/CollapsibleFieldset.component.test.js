import React from 'react';
import cases from 'jest-in-case';
import { shallow } from 'enzyme';
import CollapsiblePanel, { TYPE_ACTION } from '@talend/react-components/lib/CollapsiblePanel';
import createCollapsibleFieldset from './CollapsibleFieldset.component';

function customTitle(value, schema) {
	return `${schema.title}: ${value.firstname} ${value.lastname}`;
}

const schema = {
	title: 'Basic',
	items: [
		{
			key: ['age'],
			title: 'Age',
			schema: { type: 'number' },
			type: 'number',
		},
		{
			key: ['address'],
			title: 'Adress',
			schema: { type: 'string' },
			type: 'text',
		},
	],
};

const value = {
	firstname: 'Jimmy',
	lastname: 'Somsanith',
};

describe('CollapsibleFieldset', () => {
	cases(
		'should render',
		opts => {
			// given
			const CollapsibleFieldset = createCollapsibleFieldset(opts.titleFn);

			// when
			const wrapper = shallow(
				<CollapsibleFieldset
					id={'my-fieldset'}
					schema={schema}
					value={{ ...value, isClosed: opts.isClosed }}
				/>,
			);

			// then
			expect(wrapper.getElement()).toMatchSnapshot();
		},
		{
			'a full fieldset (header and body)': { isClosed: false },
			'a collapsed fieldset (header only)': { isClosed: true },
			'a custom title': { isClosed: false, titleFn: customTitle },
		},
	);

	it('should toggle', () => {
		// given
		const CollapsibleFieldset = createCollapsibleFieldset();
		const onChange = jest.fn();
		const event = {
			stopPropagation: jest.fn(),
			preventDefault: jest.fn(),
		};

		const wrapper = shallow(
			<CollapsibleFieldset
				id={'my-fieldset'}
				onChange={onChange}
				schema={schema}
				value={{ ...value, isClosed: true }}
			/>,
		);
		// when
		wrapper.find(CollapsiblePanel).getElement().props.onToggle(event);

		// then
		expect(event.stopPropagation).toBeCalled();
		expect(event.preventDefault).toBeCalled();
		expect(onChange).toBeCalledWith(event, {
			schema,
			value: { ...value, isClosed: false },
		});
	});

	it('should render Actions component if actions are provided', () => {
		const CollapsibleFieldset = createCollapsibleFieldset();
		const onChange = jest.fn();
		const actions = [{ id: 'action1' }, { id: 'action2' }];

		const wrapper = shallow(
			<CollapsibleFieldset
				id={'my-fieldset'}
				onChange={onChange}
				schema={schema}
				value={value}
				actions={actions}
			/>,
		);
		const header = wrapper.find(CollapsiblePanel).dive().getElement().props.header;
		expect(header.length).toBe(2);
		expect(header[1].length).toBe(2);
		expect(header[1][0].id).toBe('action1');
		expect(header[1][0].displayMode).toBe(TYPE_ACTION);
		expect(header[1][1].id).toBe('action2');
		expect(header[1][1].displayMode).toBe(TYPE_ACTION);	});

	it('should not render Actions component if actions are not provided', () => {
		const CollapsibleFieldset = createCollapsibleFieldset();
		const onChange = jest.fn();

		const wrapper = shallow(
			<CollapsibleFieldset id={'my-fieldset'} onChange={onChange} schema={schema} value={value} />,
		);

		expect(wrapper.exists('Actions')).toEqual(false);
	});
});
