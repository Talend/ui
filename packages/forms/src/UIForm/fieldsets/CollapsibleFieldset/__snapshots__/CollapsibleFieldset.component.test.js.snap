// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleFieldset should render a collapsed fieldset (header only) 1`] = `
<fieldset
  className="form-group theme-collapsible-panel collapsible-panel"
>
  <withI18nextTranslation(CollapsiblePanel)
    expanded={false}
    header={
      Array [
        Object {
          "label": "Basic",
        },
        Array [],
      ]
    }
    id="my-fieldset"
    onToggle={[Function]}
  >
    <Widget
      id="my-fieldset"
      schema={
        Object {
          "key": Array [
            "age",
          ],
          "schema": Object {
            "type": "number",
          },
          "title": "Age",
          "type": "number",
        }
      }
      value={
        Object {
          "firstname": "Jimmy",
          "isClosed": true,
          "lastname": "Somsanith",
        }
      }
      widgets={Object {}}
    />
    <Widget
      id="my-fieldset"
      schema={
        Object {
          "key": Array [
            "address",
          ],
          "schema": Object {
            "type": "string",
          },
          "title": "Adress",
          "type": "text",
        }
      }
      value={
        Object {
          "firstname": "Jimmy",
          "isClosed": true,
          "lastname": "Somsanith",
        }
      }
      widgets={Object {}}
    />
  </withI18nextTranslation(CollapsiblePanel)>
</fieldset>
`;

exports[`CollapsibleFieldset should render a custom title 1`] = `
<fieldset
  className="form-group theme-collapsible-panel collapsible-panel"
>
  <withI18nextTranslation(CollapsiblePanel)
    expanded={true}
    header={
      Array [
        Object {
          "label": "Basic: Jimmy Somsanith",
        },
        Array [],
      ]
    }
    id="my-fieldset"
    onToggle={[Function]}
  >
    <Widget
      id="my-fieldset"
      schema={
        Object {
          "key": Array [
            "age",
          ],
          "schema": Object {
            "type": "number",
          },
          "title": "Age",
          "type": "number",
        }
      }
      value={
        Object {
          "firstname": "Jimmy",
          "isClosed": false,
          "lastname": "Somsanith",
        }
      }
      widgets={Object {}}
    />
    <Widget
      id="my-fieldset"
      schema={
        Object {
          "key": Array [
            "address",
          ],
          "schema": Object {
            "type": "string",
          },
          "title": "Adress",
          "type": "text",
        }
      }
      value={
        Object {
          "firstname": "Jimmy",
          "isClosed": false,
          "lastname": "Somsanith",
        }
      }
      widgets={Object {}}
    />
  </withI18nextTranslation(CollapsiblePanel)>
</fieldset>
`;

exports[`CollapsibleFieldset should render a full fieldset (header and body) 1`] = `
<fieldset
  className="form-group theme-collapsible-panel collapsible-panel"
>
  <withI18nextTranslation(CollapsiblePanel)
    expanded={true}
    header={
      Array [
        Object {
          "label": "Basic",
        },
        Array [],
      ]
    }
    id="my-fieldset"
    onToggle={[Function]}
  >
    <Widget
      id="my-fieldset"
      schema={
        Object {
          "key": Array [
            "age",
          ],
          "schema": Object {
            "type": "number",
          },
          "title": "Age",
          "type": "number",
        }
      }
      value={
        Object {
          "firstname": "Jimmy",
          "isClosed": false,
          "lastname": "Somsanith",
        }
      }
      widgets={Object {}}
    />
    <Widget
      id="my-fieldset"
      schema={
        Object {
          "key": Array [
            "address",
          ],
          "schema": Object {
            "type": "string",
          },
          "title": "Adress",
          "type": "text",
        }
      }
      value={
        Object {
          "firstname": "Jimmy",
          "isClosed": false,
          "lastname": "Somsanith",
        }
      }
      widgets={Object {}}
    />
  </withI18nextTranslation(CollapsiblePanel)>
</fieldset>
`;
